packages/react-router-dom/modules/BrowserPrompt.js

```diff
@@ -0,0 +1,80 @@
+import React, { PropTypes } from 'react'
+import { Prompt } from 'react-router'
+
+const BEFOREUNLOAD = "beforeunload"; // see: http://stackoverflow.com/questions/39094138/reactjs-event-listener-beforeunload-added-but-not-removed
```

Comment:
I originally tried the window API "onbeforeunload" however, this did not work in React as we want unable to upload the event. Once the event was assigned, it was not possible to unregister.
https://developer.mozilla.org/en-US/docs/Web/API/WindowEventHandlers/onbeforeunload

It's force of habit to include reference to any solutions to issues we have for Sprint retros. This comment serves no purpose here. 

The comment is to show why I am using "old fashioned" string (something my team dislikes) events when there are dedicated "on" handlers built in.